src/gsCore/gsMultiPatch.hpp

```diff
@@ -240,6 +240,35 @@ void gsMultiPatch<T>::addInterface( gsGeometry<T>* g1, boxSide s1,
 }
 
 
+template<class T>
+void gsMultiPatch<T>::coordinates( const patchCorner& pc, gsMatrix<T>& coordinates )
+{
+    coordinates.resize(m_dim,1);
+    gsMatrix<T> supp = m_patches[pc.patch]->parameterRange();
+    gsVector<bool> boxPar = pc.parameters(m_dim);
+    for (index_t d=0; d<m_dim;++d)
+        coordinates(d,0) = boxPar(d) ? supp(d,1) : supp(d,0);
+}
+
+template<class T>
+void gsMultiPatch<T>::physicalCoordinates( const patchCorner& pc, gsMatrix<T>& coordinates )
+{ gsMatrix<T> tmp; gsMultiPatch<T>::coordinates(pc,tmp); m_patches[pc.patch]->eval_into(tmp,coordinates); }
+
+template<class T>
+void gsMultiPatch<T>::coordinates( const patchSide& ps, gsMatrix<T>& coordinates )
+{
+    coordinates.resize(m_dim,1);
+    gsMatrix<T> supp = m_patches[ps.patch]->parameterRange();
+    const index_t dir = ps.direction();
+    for (index_t d=0; d<m_dim;++d)
+        coordinates(d,0) = d == dir ? ( ps.parameter() ? supp(d,1) : supp(d,0) )
+                                    : ( ( supp(d,1) + supp(d,0) ) / 2 );
```

Comment:
please avoid the ternary operator in this specific case and use "if" instead (arithmetic operations can cause compilation problems with non-standard number types)